packages/x-components/src/x-modules/history-queries/store/actions/toggle-history-queries.action.ts

```diff
@@ -0,0 +1,13 @@
+import { localStorageService } from '../../../../utils/storage';
+import { HISTORY_QUERIES_ENABLED_KEY } from '../constants';
+import { HistoryQueriesXStoreModule } from '../types';
+
+export const toggleHistoryQueries: HistoryQueriesXStoreModule['actions']['toggleHistoryQueries'] =
+  async ({ dispatch, commit }, isEnabled) => {
+    if (!isEnabled) {
+      await dispatch('setHistoryQueries', []);
```

Comment:
Should we remove the key when it's not enabled?

```suggestion
      await dispatch('setHistoryQueries', []);
      localStorageService.removeItem(HISTORY_QUERIES_ENABLED_KEY);
```